client/src/features/modal/EventModal.jsx

```diff
@@ -31,15 +32,21 @@ const EventModal = ({ handleClose }) => {
       {authorCheck &&
       <button
         className="w-auto h-12 mt-5 px-2 border-solid border-2 border-gray outline-none rounded font-semibold text-xl hover:bg-teal-600 active:bg-teal-700 focus:outline-none focus:ring focus:ring-teal-300"
-        onClick={() => dataService.deleteEvent(modal.activeModal._id)}
+        onClick={() => {
+          dataService.deleteEvent(modal.activeModal._id);
+          handleClose();
+        }}
       >
         Delete Specific Event
       </button>
       }
       {authorCheck &&
       <button
         className="w-auto h-10 mt-5 px-2 border-solid border-2 border-gray outline-none rounded font-semibold text-xl text-sm hover:bg-teal-600 active:bg-teal-700 focus:outline-none focus:ring focus:ring-teal-300 inline-block"
-        onClick={() => dataService.deleteAllEvents(modal.activeModal.groupId)}
+        onClick={() => {
+          dataService.deleteAllEvents(modal.activeModal.groupId);
+          handleClose();
```

Comment:
The modal get close regardless to whether deleting the event succeed or fails. Closing the modal should occur after a sucessful deletion using the .then method.